quinn/src/send_stream.rs

```diff
@@ -283,7 +299,6 @@ impl Drop for SendStream {
         let mut conn = self.conn.state.lock("SendStream::drop");
 
         // clean up any previously registered wakers
-        conn.stopped.remove(&self.stream);
```

Comment:
Why is this removed?

Reply:
If we'd remove the `Notify` from `conn.stopped` when dropping the send stream, a future previously returned from `SendStream::stopped` would remain pending forever.
i.e.
```rust
let stopped = send_stream.stopped();
drop(send_stream);
stopped.await; // would never yield
```

The `Notify` in `conn.stopped` should be cleared when `Event::Stream(StreamEvent::Finished)` is handled in `quinn::connection::State::forward_app_events` (or at least that is my understanding, I'm not deeply invested in these codepaths yet).

Reply:
We could remove the notify here and trigger the wakeup though. I think it should be equal to notifiying from `forward_app_events` on the `StreamEvent::Finished`?

Reply:
No, I think the way you've got it now is correct -- the event should still work just fine after the stream handle is dropped.